osf/models/institution.py

```diff
@@ -18,18 +18,37 @@ class Institution(Loggable, base.ObjectIDMixin, base.BaseModel):
     # /TODO DELETE ME POST MIGRATION
 
     # TODO Remove null=True for things that shouldn't be nullable
-    banner_name = models.CharField(max_length=255, null=True, blank=True)
+
+    name = models.CharField(max_length=255)
+    description = models.TextField(blank=True, null=True, default='')
+
+    # TODO Could `banner_name` and `logo_name` be a FilePathField?
+    # Both banner_name and logo_name are the name of the file under the `banners/` or `shields/` folder in
+    # `website/static/img/institutions/`. They cannot be null or empty, otherwise OSF dashboard page and institution
+    # project page won't finish loading.
+    banner_name = models.CharField(max_length=255, blank=True, default='')
+    logo_name = models.CharField(max_length=255, blank=True, default='')
+
+    # the protocol used to delegate authentication: `CAS`, `SAML`, `OAuth`, e.t.c
+    # only institutions with a delegation protocol (not null and not empty) shows up on institution login page
+    delegation_protocol = models.CharField(max_length=255, null=True, blank=True)
```

Comment:
If this is essentially an enum you should use the [choices attribute](https://docs.djangoproject.com/en/1.10/ref/models/fields/#choices) to provide options/validation.

`CharFields` should **never** be `null=True` (after the migrations are complete). If you want to have an option where there is no delegation protocol you should add a `'None'` option to the choices field that's database value is `''`.